src/screens/Habitoptions.js

```diff
@@ -1,132 +1,145 @@
-    { id: 11, name: "Study" },
-    { id: 12, name: "Spend Time with Family" },
-    { id: 13, name: "Connect with Friends:" },
-    { id: 14, name: "Express Gratitude" },
-    { id: 15, name: "Networking" },
-    { id: 16, name: "Cooking" },
-    { id: 17, name: "Outdoor Time" },
-    { id: 18, name: "Watch a Movie" },
-    
-  ]);
   const [newHabit, setNewHabit] = useState('');
-
-  const handleSelectOption = (option) => {
-    navigation.navigate('Home', { selectedHabit: option });
-  };
+  const [expandedCategory, setExpandedCategory] = useState('');
 
   const addNewHabit = () => {
     if (newHabit.trim() === '') return;
-
-    const newId = options.length + 1;
-    setOptions([...options, { id: newId, name: newHabit }]);
+    // Add new habit to a default or specific category
+    habitCategories['Personal Growth'].push(newHabit.trim());
     setNewHabit('');
+    // Optionally, navigate to home screen with the new habit
+    navigation.navigate('Home', { selectedHabit: newHabit.trim() });
   };
-  
-  const renderOptionItem = ({ item }) => (
-    <TouchableOpacity 
-      style={styles.optionItem} 
-      onPress={() => handleSelectOption(item.name)}
-    >
-      <Text style={styles.optionText}>{item.name}</Text>
-    </TouchableOpacity>
-  );
 
   return (
-    <View style={styles.container}>
-      <Text style={styles.header}>Please Choose or Add a
-       Habit!</Text>
-
+    <ScrollView style={styles.container}>
+      <Text style={styles.header}>Please Choose or Add a Habit!</Text>
+      
       <View style={styles.inputContainer}>
         <TextInput
           style={styles.input}
           placeholder="Add a new habit"
           value={newHabit}
           onChangeText={setNewHabit}
         />
-        <TouchableOpacity style={styles.addButton} onPress={addNewHabit}>
-          <Text style={styles.addButtonText}>Add</Text>
-        </TouchableOpacity>
+        <Button title="Add" onPress={addNewHabit} />
       </View>
 
-      <FlatList
-        data={options}
-        renderItem={renderOptionItem}
-        keyExtractor={item => item.id.toString()}
-      />
-    </View>
+      {Object.entries(habitCategories).map(([category, habits]) => (
+        <View key={category}>
+          <TouchableOpacity onPress={() => setExpandedCategory(expandedCategory === category ? '' : category)}>
+            <Text style={styles.category}>{category}</Text>
+          </TouchableOpacity>
+          {expandedCategory === category && habits.map((habit, index) => (
+            <TouchableOpacity key={index} style={styles.optionItem} onPress={() => navigation.navigate('Home', { selectedHabit: habit })}>
+              <Text style={styles.optionText}>{habit}</Text>
+            </TouchableOpacity>
+          ))}
+        </View>
+      ))}
+    </ScrollView>
   );
 };
 
 const styles = StyleSheet.create({
   container: {
     flex: 1,
-    paddingTop: 20,
-    paddingHorizontal: 10,
+    backgroundColor: '#fafafa',
```

Comment:
for example, you can add comments here to explain the color of this hex

Reply:
I added comments to the hex colors